### PR TITLE
ENH: Add freezability to remaining multivariate distributions

### PR DESCRIPTION
**Pull request by NamamiShanker**
#### Reference issue

gh-15617

#### What does this implement/fix?

- [x] Implement `ortho_group` freezability
- [ ] Implement `unitary_group` freezability
- [ ] Implement `random_correlation` freezability

### CODE REVIEW

File: scipy/stats/_multivariate.py [hunk 1]
```diff
@@ -3328,6 +3328,12 @@ class special_ortho_group_gen(multi_rv_generic):
     >>> rv = special_ortho_group(5)
     >>> # Frozen object with the same methods but holding the
     >>> # dimension parameter fixed.
+    >>> rv.rvs(random_state=1234)
```
Comment (mdhaber) [inline scipy/stats/_multivariate.py:3331]: I would say it's good to add this, but there are other issues here that I don't want to get into (should use `np.random.default_rng()`, comments on the previous two lines should not be there). 
IIRC, all multivariate distributions need this same sort of treatment. Let's do that to all of them in a separate PR and limit the scope here to just adding the frozen ortho group class. Sound good?

File: scipy/stats/tests/test_multivariate.py [hunk 2]
```diff
@@ -1430,6 +1430,15 @@ def test_invalid_dim(self):
         assert_raises(ValueError, ortho_group.rvs, 1)
         assert_raises(ValueError, ortho_group.rvs, 2.5)
 
+    def test_frozen_matrix(self):
+        dim = 7
+        frozen = ortho_group(dim)
```
Comment (mdhaber) [inline scipy/stats/tests/test_multivariate.py:1435]: Should also test `ortho_group(dim, seed=1234)` and see that the first `rvs` it produces is the same as the others.

File: scipy/stats/_multivariate.py [hunk 3]
```diff
@@ -3472,6 +3478,18 @@ class ortho_group_gen(multi_rv_generic):
...
+
+    >>> rv = ortho_group(5)
+    >>> # Frozen object with the same methods but holding the
+    >>> # dimension parameter fixed.
+    >>> rv.rvs(random_state=1234)
+    array([[ 0.22707382,  0.12568963,  0.95451943, -0.04452602, -0.13982017],
+        [-0.57365132, -0.69712849,  0.25624774,  0.33489085,  0.08438747],
+        [ 0.69008482, -0.54784192, -0.02599573, -0.05657903,  0.46880188],
+        [-0.15059348,  0.44259309,  0.11106863,  0.37911176,  0.79080402],
+        [-0.34708238, -0.0468463 ,  0.10108277, -0.8596154 ,  0.35802636]])
```
Comment (mdhaber) [inline scipy/stats/_multivariate.py:3492]:
```suggestion
```
Let's put this in the same state as all the other multivariate distributions (even if it is not great).

**Review by mdhaber — COMMENTED** [review]
Looks good. Let's keep this simple, merge it, and move onto the next one. 
Please implement the change to the test, remove the `rvs` addition to the examples (we'll do it separately to all multivariate distributions), and I think we'll be set.

File: scipy/stats/_multivariate.py (hunk 1 above)
Comment (NamamiShanker) [reply scipy/stats/_multivariate.py:3331]: Yeah, that sounds good. Should I remove the previous two comment lines?

File: scipy/stats/_multivariate.py [hunk 4]
```diff
@@ -3528,6 +3541,38 @@ def rvs(self, dim, size=1, random_state=None):
...
+        """Create a frozen O(N) distribution.
+
+        Parameters
+        ----------
+        dim : scalar
+            Dimension of matrices
+        seed : {None, int, `numpy.random.Generator`,
+                `numpy.random.RandomState`}, optional
+
+            If `seed` in None (or `np.random`), the `numpy.random.RandomState`
```
Comment (mdhaber) [inline scipy/stats/_multivariate.py:3555]:
```suggestion
            If `seed` is None (or `np.random`), the `numpy.random.RandomState`
```

File: scipy/stats/_multivariate.py [hunk 5]
```diff
@@ -3528,6 +3541,38 @@ def rvs(self, dim, size=1, random_state=None):
...
+            singleton is used.
+            If `seed` in an int, a new ``RandomState`` instance is used,
```
Comment (mdhaber) [inline scipy/stats/_multivariate.py:3557]:
```suggestion
            If `seed` is an int, a new ``RandomState`` instance is used,
```

**Review by mdhaber — CHANGES_REQUESTED** [review]
Other comments: 
* Please add a `Parameters` section identical to that of `special_ortho_group_gen`. This section tells users what parameters are accepted when the class is used as a callable. (This wasn't there before because `ortho_group` was not callable, but it is needed due to the changes in this PR.)
* Not required, but would be nice: add a "See Also" section similar to `special_ortho_group`, listing at least `special_ortho_group`. I am not sure if `scipy.spatial.transform.Rotation.random` is still appropriate (because I haven't read into what this is doing).
* I would move the `__call__` method of `ortho_group` before the `_process_parameters` method just to match `special_ortho_group`, if you don't mind.

Otherwise, this looks good to me. Once these changes are made, I think I can merge.

**Comment by NamamiShanker** [comment]
>  I am not sure if scipy.spatial.transform.Rotation.random is still appropriate (because I haven't read into what this is doing).

`special_ortho_group_gen` returns a random Rotation matrix, which is relevant to `scipy.spatial.transform.Rotation.random`. I don't think `ortho_group_gen` is relevant because it just returns a random orthogonal matrix.

**Review by mdhaber — APPROVED** [review]
I'll merge when tests are green.

**Pull request merged**